modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/js/KBDropdownPropsTransformer.js

```diff
@@ -28,12 +42,87 @@ const ACTIONS = {
 		});
 	},
 
-	move({moveItemUrl}) {
-		openModal({
+	move(
+		{
+			moveItemActionUrl,
+			moveItemModalUrl,
+			selectedItemClassNameId,
+			selectedItemId,
+			selectedItemType,
+		},
+		portletNamespace
+	) {
+		openSelectionModal({
+			buttonAddLabel: Liferay.Language.get('move'),
 			height: '50vh',
+			multiple: true,
+			onSelect: ({index, item, parentItem}) => {
+				if (!item) {
+					item = {
+						classNameId: selectedItemClassNameId,
+						id: selectedItemId,
+						type: selectedItemType,
+					};
+				}
+
+				if (
+					item.type === ITEM_TYPES.folder &&
+					parentItem.type === ITEM_TYPES.article
+				) {
+					openToast({
+						message: Liferay.Language.get(
+							'folders-cannot-be-moved-into-articles'
+						),
+						type: 'danger',
+					});
+
+					return false;
+				}
+
+				fetch(moveItemActionUrl, {
+					body: objectToFormData({
+						[`${portletNamespace}dragAndDrop`]: true,
+						[`${portletNamespace}position`]: index?.next ?? -1,
+						[`${portletNamespace}resourceClassNameId`]: item.classNameId,
```

Comment:
for `resourceClassNameId` and `resourcePrimKey` use the data you have from args.